tests/test_etl.py

```diff
@@ -28,7 +29,17 @@ def test_all_data_steps_have_code():
         assert s.can_execute(), f'no code found for step "data://{s.path}"'
 
 
-def get_all_steps() -> List[Step]:
-    dag = load_dag()
+def test_sub_dag_import():
+    # Ensure that sub-dag is imported from separate file
+    sub_dag_step = "general_files"
+    assert [
+        s
+        for s in get_all_steps(filename="tests/data/dag.yml")
```

Comment:
Nitpick - to make our lives simpler I'd only test `load_dag` here. That way you don't need real steps as you won't be calling `compile_steps` and you only test the function of interest.

Reply:
Good call. That also allows me to make the test data generic because I don't have to worry about finding a recipe to execute the step.